web-common/src/features/sources/source-imported-utils.ts

```diff
@@ -13,12 +14,75 @@ export async function checkSourceImported(
     fileArtifacts.getFileArtifact(filePath).lastStateUpdatedOn;
   if (lastUpdatedOn) return; // For now only show for fresh sources
 
-  const success = await waitForResourceUpdate(
-    queryClient,
-    get(runtime).instanceId,
-    filePath,
-  );
+  waitForResourceUpdate(queryClient, get(runtime).instanceId, filePath)
+    .then((success) => {
+      console.log("s", success);
+      if (!success) return;
+      sourceImportedPath.set(filePath);
```

Comment:
(In the future) this could be a good use case for the Event Bus. Showing an overlay in response to an event is very analogous to showing a toast notification in response to an event. @briangregoryholmes has recently cleaned up the `NotificationCenter.svelte` – I'd recommend checking out that new code.

Reply:
Hmmm good point. Will do a follow up after the realse.